### DISCUSSION TITLE
regression testing

### DISCUSSION THREAD

**Discussion by boredland** [Manjaro-Sway/manjaro-sway#207] [Ideas]
@codesardine and I were discussion if and how to implement some kind of regression testing, on the "built image" level, but especially for package updates. I suggest we collect "things we could test" as threads here and each discuss how / what we could test.

**Comment by boredland** [comment]
package update: is it still installable on all editions?

**Reply by codesardine** [reply]
I assume you mean editions?

I think having a build that is not DE dependable could be useful to test the base.

**Comment by boredland** [comment]
package update: is it still installable given package x is installed on the machine?

**Reply by codesardine** [reply]
I think we can add is install able and runs without errors.

**Comment by codesardine** [comment]
package update: does package X updates without conflicts?

**Reply by boredland** [reply]
I guess this could be the easiest one and perhaps I already do this in our [packagebuilds](https://github.com/Manjaro-Sway/manjaro-sway-settings-git/blob/master/.github/workflows/pkgbuild.yml#L80) and would basically only need to add "i" to check if its installable (or some dry run, if that exists).

**Reply by boredland** [reply]
may I ask: how is this done in Boxit @philmmanjaro @oberon-manjaro (or whoever)? what is the chroot there based on?

**Reply by hphilm** [reply]
Boxit is just a server management software and doesn't build packages. We have chrootbuild for that purpose.

**Reply by boredland** [reply]
ah, I thought chrootbuild was part of it. so what kind of chroot does chrootbuild build in?

**Comment by boredland** [comment]
perhaps the more complicated one (but the one I miss the most):
the live iso boots straight to desktop and I can open a terminal. I imagine sth. like cypress/selenium, but for qemu.

**Reply by codesardine** [reply]
for graphical tests as far as i now there is only http://open.qa/ by suse, a server is needed and I will take some considerable time to write graphical tests.

I was thinking more we launch a command or application return the process output, if it has crashed or not, that should be sufficient to catch regressions.

**Reply by hphilm** [reply]
Check out this video from the QGIS project. They talk about test organization and where Kiwi TCMS fits into their infrastructure:
https://www.youtube.com/watch?v=Wob_L_SrEwE

**Reply by codesardine** [reply]
This provides integration between Github and Kwie CMS test framework, most manjaro things are hosted on gitlab, also this runs some python unittests that is what i mentioned above, we need to decide whats going to be tested and what kind of tests we doing.

**Reply by boredland** [reply]
> This provides integration between Github and Kwie CMS test framework, most manjaro things are hosted on gitlab.

wdym with "This"? qgis or open.qa?

**Reply by codesardine** [reply]
I never used any of them, but open.qa is only Graphical tests if I am not mistaken, not qgis -> https://kiwitcms.readthedocs.io/en/latest/about.html is what qgis uses.

**Reply by boredland** [reply]
Me neither. Ideally or a first step would be some kind of visual testing for me, e.g., after 2minutes of qemu booting, take a screenshot and verify it looks how it is supposed to.

If it doesn't, create a PR with the new image, so we can verify if it is supposed to be like that or a regression.

**Reply by codesardine** [reply]
Seems that it can be run in a container https://open.qa/docs/#_openqa_in_a_container ideally we might need a new server @philmmanjaro Is this something you want?

@boredland as for creating tests with openqa we have to take some time reading the docs.

**Reply by codesardine** [reply]
Or maybe we can try this locally first to access if it is viable or not.

**Reply by boredland** [reply]
ya, we should. I don't know if I have time right now, but would be happy to assist / try this together.

**Reply by codesardine** [reply]
I am also busy at the moment, but will put this in my todo list, if I forget remind me in a couple of weeks.
